### DISCUSSION TITLE
Rendering Document Editor

### DISCUSSION THREAD

**Discussion by yash-93** [keystonejs/keystone#5488] [Feature requests]
I want to render Document editor as done here https://keystonejs.com/docs/guides/document-field-demo, in a NextJS application. What is the possible way to do so?

**Comment by gautamsi** [comment]
@mitchellhamilton can you provide any help here please

**Comment by emmatown** [comment]
Currently, we intentionally don't expose the ability to render the document editor outside of Keystone, we likely eventually will but we want to spend time to figure out the right abstraction before we do.

**Reply by adeperio** [reply]
Hi @mitchellhamilton just seeing if this is still an issue? I'm trying to integrate this in a NextJS client web app at the moment. I guess it's not supported outside of Keystone? It would be great to have the editor available. Could we use Slate's editor to read the document's content?

**Reply by anthony-sap** [reply]
I'd also like to know if a standalone slate editor instance can read the content that the Keystone version of slate creates?

**Comment by willemmulder** [comment]
In the meantime, you might want to try this: https://gist.github.com/sladg/c0fc87e82c8fbc6fe5488f5de5070296 to render a standalone editor.

**Reply by lilac** [reply]
I also want to use the document editor stand-alone, as a beautiful example of the slate editor. Thanks for your great work!
Now I've implemented a PoC [here](https://github.com/in-fun/fun-editor).
As for the keystone design and abstraction, I think it can be separated as a package, just like the `document-renderer`. That is, we'll have 3 packages.
1. `document-renderer` for rendering content.
2. `document-editor` depends on the renderer.
3. `fields-document` depends on the `document-editor`.

@mitchellhamilton What do you think? If you like the idea, I may find some time to contribute it.

**Comment by mingfang** [comment]
I figured out how to package the document-editor.
https://github.com/mingfang/keystone/commit/e62a4d7d762e5ddfd2d67421c82facedd17e2734

I can now use the document-editor in my own app.

**Reply by rahulp18** [reply]
Yeah it's easy use slate.js it is compactable with keystone document field
